addons/website_forum/models/forum_forum.py

```diff
@@ -131,11 +134,50 @@ def _get_default_welcome_message(self):
 
     can_moderate = fields.Boolean(string="Is a moderator", compute="_compute_can_moderate")
 
+    # tags
+    tag_ids = fields.One2many('forum.tag', 'forum_id', string='Tags')
+    tag_most_used_ids = fields.One2many('forum.tag', compute='_compute_tag_ids_usage')
+    tag_unused_ids = fields.One2many('forum.tag', compute='_compute_tag_ids_usage')
+
     @api.depends_context('uid')
     def _compute_can_moderate(self):
         for forum in self:
             forum.can_moderate = self.env.user.karma > forum.karma_moderate
 
+    @api.depends('post_ids', 'post_ids.tag_ids', 'post_ids.tag_ids.posts_count', 'tag_ids')
+    def _compute_tag_ids_usage(self):
+        if not self.tag_ids:
+            self.tag_unused_ids = self.tag_most_used_ids = False
+            return
+
+        query = """
+            SELECT id, forum_id, posts_count
+              FROM forum_tag
+             WHERE id IN %(tag_ids)s
+          ORDER BY forum_id, posts_count DESC, name
```

Comment:
just to make it deterministic

```suggestion
          ORDER BY forum_id, posts_count DESC, name, id
```